### DISCUSSION TITLE
docker stopped working overnight

### DISCUSSION THREAD

**Discussion by danlwright** [haugene/docker-transmission-openvpn#2443] [Q&A]
I have been running the same docker container for a while without issue, but a couple of nights ago it stopped working. I had the container set to automatically restart. I'm not sure if NordVPN blocked me because of all the attempted requests.

Here is what my logs show when I attempt to start the container:

Starting container with revision: b33d0fe4c938259a0d4eb844e55468f387456121
Creating TUN device /dev/net/tun
Using OpenVPN provider: NORDVPN
Running with VPN_CONFIG_SOURCE auto
Provider NORDVPN has a bundled setup script. Defaulting to internal config
Executing setup script for NORDVPN
INFO: OVPN: Checking curl installation
INFO: OVPN: DNS resolution ok

Can someone get me pointed in the right direction to resolve the issue?

Thanks,

Dan

**Comment by edgd1er** [comment]
Nordvpn throttle requests to the api if that's occuring, you should have a log saying that html was found in api response.
You may check if your ip is throttled through a simple light request: https://api.nordvpn.com/v1/servers/groups
```
[
  {
    "id": 1,
    "created_at": "2017-06-13 13:41:00",
    "updated_at": "2017-06-13 13:41:00",
    "title": "Double VPN",
    "identifier": "legacy_double_vpn",
    "type": {
      "id": 3,
      "created_at": "2017-06-13 13:40:17",
      "updated_at": "2017-06-13 13:40:23",
      "title": "Legacy category",
      "identifier": "legacy_group_category"
    }
  },....
```

If you are blacklisted, stop the container, a few hours later the API would be available again.
I think there is a NORDVPN_SERVER that by pass all api request and get directly the ovpn file configuration. this could be a workaround during the api's throttle.

**Reply by danlwright** [reply]
I guess that I must not be blacklisted. I had the container stopped and the restart policy set to none, so that it wouldn't keep attempting to start. I just tried starting it again and got the same log results. Any other ideas?

Thanks!

**Comment by edgd1er** [comment]
DEBUG env var if set to TRUE will activate -x flag in bash, allowing verbose mode.
You may set the var and post the log. the previously posted log show no error and is incomplete.

**Reply by danlwright** [reply]
Thanks @edgd1er that definitely yielded more output. Hopefully it will mean more to you than it does to me.

Here is the output:

+ [[ -n b33d0fe4c938259a0d4eb844e55468f387456121 ]]
+ echo 'Starting container with revision: b33d0fe4c938259a0d4eb844e55468f387456121'
+ [[ -x /scripts/openvpn-pre-start.sh ]]
Starting container with revision: b33d0fe4c938259a0d4eb844e55468f387456121
+ grep -q OVERRIDE_DNS

+ compgen -e
+ nslookup google.com
Creating TUN device /dev/net/tun
+ [[ true == \t\r\u\e ]]
+ echo 'Creating TUN device /dev/net/tun'
+ rm -f /dev/net/tun
+ mkdir -p /dev/net
+ mknod /dev/net/tun c 10 200
+ chmod 0666 /dev/net/tun
+ VPN_PROVIDER=NORDVPN
+ export VPN_PROVIDER=nordvpn
+ VPN_PROVIDER=nordvpn
+ export VPN_PROVIDER_HOME=/etc/openvpn/nordvpn
+ VPN_PROVIDER_HOME=/etc/openvpn/nordvpn
+ mkdir -p /etc/openvpn/nordvpn
Using OpenVPN provider: NORDVPN
+ [[ -z '' ]]
+ [[ NORDVPN == \*\*\N\o\n\e\*\* ]]
+ [[ -z NORDVPN ]]
+ echo 'Using OpenVPN provider: NORDVPN'
+ [[ nordvpn == \c\u\s\t\o\m ]]
+ [[ -n '' ]]
+ [[ -z '' ]]
+ VPN_CONFIG_SOURCE=auto
+ VPN_CONFIG_SOURCE=auto
+ echo 'Running with VPN_CONFIG_SOURCE auto'
+ [[ auto == \a\u\t\o ]]
+ [[ -x /etc/openvpn/nordvpn/configure-openvpn.sh ]]
+ echo 'Provider NORDVPN has a bundled setup script. Defaulting to internal config'
Running with VPN_CONFIG_SOURCE auto
Provider NORDVPN has a bundled setup script. Defaulting to internal config
+ VPN_CONFIG_SOURCE=internal
+ [[ internal == \e\x\t\e\r\n\a\l ]]
+ [[ -x /etc/openvpn/nordvpn/configure-openvpn.sh ]]
+ echo 'Executing setup script for NORDVPN'

+ pushd -n /
+ . /etc/openvpn/nordvpn/configure-openvpn.sh
++ set -e -u -o pipefail
++ MAIN_DIR=/etc/openvpn/..
++ [[ -f /etc/openvpn/../utils.sh ]]
++ true
++ nordvpn_api=https://api.nordvpn.com
++ nordvpn_dl=downloads.nordcdn.com
++ nordvpn_cdn=https://downloads.nordcdn.com/configs/files
++ nordvpn_doc=https://haugene.github.io/docker-transmission-openvpn/provider-specific/#nordvpn
++ possible_protocol='tcp, udp'
++ VPN_PROVIDER_HOME=/etc/openvpn/nordvpn
++ NORDVPN_COUNTRY=US
++ NORDVPN_CATEGORY=legacy_p2p
++ NORDVPN_PROTOCOL=udp
++ NORDVPN_SERVER=
++ cd /etc/openvpn
++ script_init
++ log 'INFO: OVPN: Checking curl installation'
++ printf '%b\n' 'INFO: OVPN: Checking curl installation'
INFO: OVPN: Checking curl installation
++ script_needs curl
++ command -v curl
++ checkDNS
Executing setup script for NORDVPN
+++ dig +short downloads.nordcdn.com
++ res='104.17.168.30
104.17.167.30'
++ '[' -z '104.17.168.30
104.17.167.30' ']'
++ log 'INFO: OVPN: DNS resolution ok'
++ printf '%b\n' 'INFO: OVPN: DNS resolution ok'
INFO: OVPN: DNS resolution ok
++ ping -c2 downloads.nordcdn.com

**Reply by edgd1er** [reply]
is the log complete ? the api was not contacted. if the log is complete, it seems that your system is killing the container few seconds after starting it.

**Reply by danlwright** [reply]
Yep, that is complete. I've tried deleted the config files and deploying a new stack from scratch (thinking that maybe something had gotten corrupted), but still have the same results.

Any ideas what might be causing it or where to dig for more info?

Thanks

**Reply by danlwright** [reply]
Come to think of it, I had some stability issues with Docker and had to revert back to version 4.12.0. However, Haugene ran successfully for a while after that. But, maybe it only ran successfully until it needed to restart. I don't know exactly how the two different Docker versions would have affected the startup process of the container - any thoughts on where/what to look for?

Thanks again!

**Reply by edgd1er** [reply]
you should check your host as it seems the container is killed. `docker ps` will show running containers.
There is a [debug](https://haugene.github.io/docker-transmission-openvpn/debug/) page in Haugene's documentation that may help you.

**Reply by danlwright** [reply]
Thanks so much. I was able to run through the troubleshooting steps and get it working.

**Reply by edgd1er** [reply]
@danlwright , if someone is having a similar problem, could you tell us how you fix it ?

**Reply by danlwright** [reply]
I will post the resolution once I get it figured out. It stopped running during the same night that I got it running. Haven't been able to get it working yet.

**Reply by danlwright** [reply]
Although I haven't found a solution, I have been collecting more information. This is what the log shows prior to the container exiting:

2022-12-05 00:00:20 VERIFY OK: depth=2, C=PA, O=NordVPN, CN=NordVPN Root CA
2022-12-05 00:00:20 VERIFY OK: depth=1, C=PA, O=NordVPN, CN=NordVPN CA7
2022-12-05 00:00:20 VERIFY KU OK
2022-12-05 00:00:20 Validating certificate extended key usage
2022-12-05 00:00:20 ++ Certificate has EKU (str) TLS Web Server Authentication, expects TLS Web Server Authentication
2022-12-05 00:00:20 VERIFY EKU OK
2022-12-05 00:00:20 VERIFY X509NAME OK: CN=us9949.nordvpn.com
2022-12-05 00:00:20 VERIFY OK: depth=0, CN=us9949.nordvpn.com
2022-12-05 00:00:20 Outgoing Data Channel: Cipher 'AES-256-GCM' initialized with 256 bit key
2022-12-05 00:00:20 Incoming Data Channel: Cipher 'AES-256-GCM' initialized with 256 bit key
2022-12-05 00:00:20 Control Channel: TLSv1.3, cipher TLSv1.3 TLS_AES_256_GCM_SHA384, peer certificate: 4096 bit RSA, signature: RSA-SHA512
2022-12-05 00:57:50 VERIFY OK: depth=2, C=PA, O=NordVPN, CN=NordVPN Root CA
2022-12-05 00:57:50 VERIFY OK: depth=1, C=PA, O=NordVPN, CN=NordVPN CA7
2022-12-05 00:57:50 VERIFY KU OK
2022-12-05 00:57:50 Validating certificate extended key usage
2022-12-05 00:57:50 ++ Certificate has EKU (str) TLS Web Server Authentication, expects TLS Web Server Authentication
2022-12-05 00:57:50 VERIFY EKU OK
2022-12-05 00:57:50 VERIFY X509NAME OK: CN=us9949.nordvpn.com
2022-12-05 00:57:50 VERIFY OK: depth=0, CN=us9949.nordvpn.com
2022-12-05 00:57:50 Outgoing Data Channel: Cipher 'AES-256-GCM' initialized with 256 bit key
2022-12-05 00:57:50 Incoming Data Channel: Cipher 'AES-256-GCM' initialized with 256 bit key
2022-12-05 00:57:50 Control Channel: TLSv1.3, cipher TLSv1.3 TLS_AES_256_GCM_SHA384, peer certificate: 4096 bit RSA, signature: RSA-SHA512
2022-12-05 01:53:13 [us9949.nordvpn.com] Inactivity timeout (--ping-restart), restarting
2022-12-05 01:53:13 SIGTERM received, sending exit notification to peer
2022-12-05 01:53:14 /etc/openvpn/tunnelDown.sh tun0 1500 1585 10.8.3.5 255.255.255.0 init
resolv.conf was restored
Sending kill signal to transmission-daemon
Waiting 5s for transmission-daemon to die
Successfuly closed transmission-daemon
2022-12-05 01:53:18 net_route_v4_del: 193.29.61.16/32 via 172.18.0.1 dev [NULL] table 0 metric -1
2022-12-05 01:53:18 net_route_v4_del: 0.0.0.0/1 via 10.8.3.1 dev [NULL] table 0 metric -1
2022-12-05 01:53:18 net_route_v4_del: 128.0.0.0/1 via 10.8.3.1 dev [NULL] table 0 metric -1
2022-12-05 01:53:18 Closing TUN/TAP interface
2022-12-05 01:53:18 net_addr_v4_del: 10.8.3.5 dev tun0
2022-12-05 01:53:18 SIGTERM[soft,exit-with-notification] received, process exiting

I do have an environment variable set for: OPENVPN_OPTS
--inactive 3600 --ping 10 --ping-exit 60, but that doesn't seem to be keeping the connection alive.

Once the container exits, it cannot be restarted as usual. Even deleting the container and recreating it via docker compose doesn't work. Restarting Docker also has no effect. The only thing that I have found that consistently allows the container to be restarted is rebooting the machine.
